hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/health/ECReplicationCheckHandler.java

```diff
@@ -119,7 +132,7 @@ public ContainerHealthResult checkHealth(ContainerCheckRequest request) {
       return new ContainerHealthResult.UnderReplicatedHealthResult(
           container, remainingRedundancy, dueToDecommission,
           replicaCount.isSufficientlyReplicated(true),
-          replicaCount.isUnrecoverable());
+          replicaCount.isUnrecoverable(), placementStatus);
```

Comment:
In the RatisReplicationCheckHandler, I added two fields to the `UnderReplicatedHealthResult`, for `isMisReplicated` and `isMisReplicatedAfterPending` - It would be better if we could use those rather than adding the placementStatus to the `UnderReplicatedHealthResult` object.

At the very least we should be consistent in the approach between the Ratis handler and this, as they both use that same `UnderReplicatedHealthResult` object and we could have common code later processing mis-replication from both Ratis and EC together, as fixing it just involved moving containers to a new rack usually.

Also have a check in `RatisReplicationCheckHandler#handle()` it increments a counter on the report object for each mis-replicated container, so we need to add that logic here too.